packages/riverpod/lib/src/framework/provider_base.dart

```diff
@@ -884,11 +907,24 @@ The provider ${_debugCurrentlyBuildingElement!.origin} modified $origin while bu
     _listeners.clear();
   }
 
+  /// Triggered when the listeners count changes. Runs right after [_onListen]
+  /// and [_onRemoveListener]
+  void _notifyListenersChanged() {
+    assert(() {
+      RiverpodBinding.debugInstance.providerChanged(
```

Comment:
Why do we notify that a provider has changed when we add/remove listeners?

Reply:
to be able to show the alert saying that the provider might be outdated. Like refreshing the list when this event happens. 

We could just remove it, it would make the implementation smaller on both sides, but the displayed data would be less realtime. 

Maybe adding a "refresh all containers" (I mean only on the devtools side, not refreshing the providers states) button would be better, this way we could refresh the devtools data. WDYT?

Reply:
But why "providerChanged"? I would expect a new event for this.
We can create as many events as we want. No need to reuse the same one for multiple things

Reply:
oh! Got it! Well, in the devtools implementation they would do the same thing, which is to refresh the provider state. That is why I used the same event, but I could change it to something like `'riverpod:provider_listeners_changed'` and listen to both events there on devtools.